utils/mapper/mapper.go

```diff
@@ -0,0 +1,40 @@
+package mapper
+
+import (
+	"errors"
+	"soarca/models/cacao"
+)
+
+func Variables(variablesInScope cacao.Variables,
+	outArgs []string,
+	returnedVariables cacao.Variables,
+	capabilityReturnArgs []string) (cacao.Variables, error) {
+
+	if len(outArgs) == 0 {
+		return cacao.NewVariables(), nil
+	}
+
+	if len(outArgs) != len(capabilityReturnArgs) {
+		return cacao.NewVariables(), errors.New("number of outargs does not match data array length")
+	}
+
+	outputVariables := cacao.NewVariables()
+
+	for i, arg := range outArgs {
+		variable, found := variablesInScope.Find(arg)
+		if !found {
+			return cacao.NewVariables(), errors.New("key is not found in variables")
+		}
+		dataVariable, ok := returnedVariables.Find(capabilityReturnArgs[i])
```

Comment:
This logic assumes that:
- a capability/command can return more than one arg separately (which might not be the case if we indeed can practically just assign the output of a command to one variable)
- the playbook implementer knows how many arguments the invocation of a command (capability) returns,
- the arguments returned from the command (capability) are in an order consistent with the order of the variables in out_args

If we're fine with these assumptions, we should at least mention this functioning in the documentation.

Reply:
These are documented in modules